### DISCUSSION TITLE
Prevent BepinEX folder deletion during application startup

### DISCUSSION THREAD

**Discussion by nkiehne** [BepInEx/BepInEx#814] [Help and Q&A]
Hi there, I am facing an issue with an application that checks for unknown files in its root directory and deletes them.
Naturally, I can't get BepinEX to work then...

Looking for any ideas and recommendations!

**Comment by mygamingaccount** [comment]
Have you tried to fix it with BepInEx by patching whatever deletes the files?

**Reply by nkiehne** [reply]
The problem is that bepinex gets deleted before it can setup anything. So unfortunately, I can't patch it out that way...

**Reply by toebeann** [reply]
> The problem is that bepinex gets deleted before it can setup anything.

Plugins do, perhaps. But patchers? Doubtful. They run before any game code is even loaded.

BepInEx 5: https://docs.bepinex.dev/articles/dev_guide/preloader_patchers.html

BepInEx 6: https://docs.bepinex.dev/v6.0.0-pre.1/articles/dev_guide/preloader_patchers.html

If you use dnSpy to find what game method is deleting BepInEx, you can then patch it away with Mono.Cecil, or even harmony patch it.

Do note that you will likely need to a search for the correct method to patch based on the method's IL content, as they will likely move it around in future updates in an attempt to break your patcher again. So you will not be able to rely on hard-coded method names unless you are happy to update the patcher every time the game updates. You are in a war of attrition with the game dev.
